contracts/programs/gummyroll/src/lib.rs

```diff
@@ -296,6 +299,7 @@ pub mod gummyroll {
         let (roll_bytes, canopy_bytes) = rest.split_at_mut(merkle_roll_size);
         let id = ctx.accounts.merkle_roll.key();
         let change_log = merkle_roll_apply_fn!(header, id, roll_bytes, initialize,)?;
+        wrap_event(change_log.try_to_vec()?, &ctx.accounts.candy_wrapper)?;
         emit!(*change_log);
```

Comment:
Just restating here:

The point of having BOTH event data in logs & in CPI args saved to transaction information is to allow for log-based indexing, correct?

i.e. if logs are truncated for appends, we would have to replay the transaction(s) to figure out the order in which the new NFTs were appended. However this requires maintaining state while parsing (to figure out the order of appends), which is undesirable at scale. This is why we emit changelogs in the WRAP cpi args -- to allow for stateless parsing, correct?

Reply:
The rationale is that we don't want the existing system to break just yet. Common keep both paradigms and deprecate when everything has been migrated